### PR TITLE
add a switch and a function to get mime-headers

### PR DESCRIPTION
**Pull request by r10s**
- add a new function to get the mime-headers of messages, currently incoming only, dc_get_mime_headers()
- to collect the headers so that the function can word, one has to set the flag `save_mime_headers` using dc_set_config()

in contrast to my proposal in #365, i save the headers in a database as i do not want to mix files of too many different sources in the blob directory.

### CODE REVIEW

File: src/dc_context.c [hunk 1]
```diff
@@ -376,6 +376,7 @@ static char* get_sys_config_str(const char* key)
  * - `e2ee_enabled` = 0=no end-to-end-encryption, 1=prefer end-to-end-encryption (default)
  * - `mdns_enabled` = 0=do not send or request read receipts,
  *                    1=send and request read receipts
+ * - `save_mime_headers` = set this to "1" if you want to use dc_get_mime_headers() later
```
Comment (hpk42) [inline src/dc_context.c]: wasn't there a list somewhere returned by ``sys.config_keys``?

Comment (r10s) [reply src/dc_context.c]: f*ck ...

Comment (hpk42) [reply src/dc_context.c]: you warned against it ;)

File: python/src/deltachat/account.py [hunk 2]
```diff
@@ -222,6 +223,25 @@ def get_message_by_id(self, msg_id):
...
+
+        :param msg_id: integer message id
+        :returns: email-mime message object.
+        """
+        import email.parser
+        mime_headers = lib.dc_get_mime_headers(self._dc_context, msg_id)
+        if mime_headers:
+            s = ffi.string(mime_headers)
+            if isinstance(s, bytes):
+                s = s.decode("ascii")
```
Comment (hpk42) [inline python/src/deltachat/account.py]: not sure if this is fully correct but it works in my setup. i think the char-string returned from dc_get_mime_headers is an ascii-string -- unicode chars in mime-header values have a special encoding (dealt with by the email parsing message object which this function returns)

**Comment by hpk42** [comment]
not sure -- headers are a bit special -- they don't use "utf8" encoding
but some odd other encoding (would have to look up ...) that is expressed
with 7-bit chars.  This is all quite different from the encoding
of the mime-body/payload.

**Pull request merged**